### PR TITLE
Add a doc for AWS Neuron in Diffusers

### PR DESCRIPTION
**Pull request by JingyaHuang**
# What does this PR do?

Add a doc linked to Optimum Neuron to demonstrate how to run diffusers on AWS Neuron devices.

### CODE REVIEW

**Comment by JingyaHuang** [comment]
Gently pinging @sayakpaul for review 🤗 .

**Review by stevhliu — APPROVED** [review]
Super nice, thanks for adding!

I think a little more explanation or description of what Optimum Neuron (can just be a sentence or two) is would be clearer for users that Optimum Neuron is how you should interface with AWS Neuron.

**Comment by JingyaHuang** [comment]
Thanks @stevhliu for the review and suggestions! I added a brief introduction to Optimum Neuron.

File: docs/source/en/optimization/neuron.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# AWS Neuron
+
+Diffusers functionalities are available on [AWS Inf2 instances](https://aws.amazon.com/ec2/instance-types/inf2/), which are EC2 instances powered by [Neuron machine learning accelerators](https://aws.amazon.com/machine-learning/inferentia/). These instances aim to provide better compute performance (higher throughput, lower latency) with good cost-efficiency, making them good candidates for AWS users to deploy diffusion models to production.
+
+[Optimum Neuron](https://huggingface.co/docs/optimum-neuron/en/index) is the interface between 🤗 HugiingFace libraries and AWS Accelerators including AWS [Trainium](https://aws.amazon.com/machine-learning/trainium/) and AWS [Inferentia](https://aws.amazon.com/machine-learning/inferentia/). It supports many of the features in Diffusers with similar APIs, so it is easier to learn if you're already familiar with Diffusers. Once you have created an AWS Inf2 instance, install Optimum Neuron.
```
Comment (stevhliu) [inline docs/source/en/optimization/neuron.md:17]:
```suggestion
[Optimum Neuron](https://huggingface.co/docs/optimum-neuron/en/index) is the interface between Hugging Face libraries and AWS Accelerators, including AWS [Trainium](https://aws.amazon.com/machine-learning/trainium/) and AWS [Inferentia](https://aws.amazon.com/machine-learning/inferentia/). It supports many of the features in Diffusers with similar APIs, so it is easier to learn if you're already familiar with Diffusers. Once you have created an AWS Inf2 instance, install Optimum Neuron.
```

**Review by stevhliu — APPROVED** [review]
One more typo and then we can merge!

File: docs/source/en/optimization/neuron.md (hunk 1 above)
Comment (JingyaHuang) [reply docs/source/en/optimization/neuron.md]: Good catch! Fixed!

**Pull request merged**